src/effects/backends/builtin/compressoreffect.cpp

```diff
@@ -0,0 +1,215 @@
+#include "effects/backends/builtin/compressoreffect.h"
+
+// static
+QString CompressorEffect::getId() {
+    return "org.mixxx.effects.compressor";
+}
+
+//static
+EffectManifestPointer CompressorEffect::getManifest() {
+    EffectManifestPointer pManifest(new EffectManifest);
+    pManifest->setId(getId());
+    pManifest->setName(QObject::tr("Compressor"));
+    pManifest->setShortName(QObject::tr("Compressor"));
+    pManifest->setAuthor("The Mixxx Team");
+    pManifest->setVersion("1.0");
+    pManifest->setDescription("A single-band compressor effect");
+    pManifest->setEffectRampsFromDry(true);
+    pManifest->setMetaknobDefault(0.0);
+
+    EffectManifestParameterPointer clipping = pManifest->addParameter();
+    clipping->setId("clipping");
+    clipping->setName(QObject::tr("Clipping"));
+    clipping->setShortName(QObject::tr("Clipping"));
+    clipping->setDescription(QObject::tr("Hard limiter to 0 db"));
+    clipping->setValueScaler(EffectManifestParameter::ValueScaler::Toggle);
+    clipping->setRange(0, 1, 1);
+    clipping->appendStep(qMakePair(QObject::tr("Off"), Clipping::ClippingOff));
+    clipping->appendStep(qMakePair(QObject::tr("On"), Clipping::ClippingOn));
+
+    EffectManifestParameterPointer autoMakeUp = pManifest->addParameter();
+    autoMakeUp->setId("automakeup");
+    autoMakeUp->setName(QObject::tr("Auto Makeup Gain"));
+    autoMakeUp->setShortName(QObject::tr("Makeup"));
+    autoMakeUp->setDescription(QObject::tr(
+            "The AutoMakeup button enables automatic makeup gain to 0 db level"));
```

Comment:
In fact this description isn't true, because this algorithm adjusts volume to 0 dBFS (actually around -3 dBFS) in anyways. So it doesn't matter, what was the original level, output will be almost 0 dBFS (even if original input was -30 dBFS for example).

Reply:
OK, than you may add a true description?

Reply:
I think originally it was true, however your suggestion changes the meaning. If you don't like the original one, I can try to reformulate it. In fact it just adjusts volume to almost 0 dBFS level for any input.

Reply:
I was thinking about it and it's just come to my mind that maybe it's better to call it "AGC (Auto Gain Coontrol)"? What do you think?